eclair-core/src/main/scala/fr/acinq/eclair/api/NewService.scala

```diff
@@ -0,0 +1,251 @@
+import akka.http.scaladsl.server.directives.{Credentials, LoggingMagnet}
+import akka.stream.{ActorMaterializer, OverflowStrategy}
+import akka.stream.scaladsl.{BroadcastHub, Flow, Keep, Source}
+import fr.acinq.eclair.payment.{PaymentLifecycle, PaymentReceived, PaymentRequest}
+import grizzled.slf4j.Logging
+import scodec.bits.ByteVector
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+trait NewService extends Directives with Logging with MetaService {
+
+  // important! Must NOT import the unmarshaller as it is too generic...see https://github.com/akka/akka-http/issues/541
+  import JsonSupport.marshaller
+  import JsonSupport.formats
+  import JsonSupport.serialization
+
+  def password: String
+
+  def eclairApi: EclairApi
+
+  implicit val actorSystem: ActorSystem
+  implicit lazy val ec = actorSystem.dispatcher
+  implicit val mat: ActorMaterializer
+
+  // a named and typed URL parameter used across several routes, 32-bytes hex-encoded
+  val channelIdNamedParameter = "channelId".as[ByteVector32](sha256HashUnmarshaller)
+  val shortChannelIdNamedParameter = "shortChannelId".as[ShortChannelId](shortChannelIdUnmarshaller)
+
+  val apiExceptionHandler = ExceptionHandler {
+    case t: Throwable =>
+      logger.error(s"API call failed with cause=${t.getMessage}", t)
+      complete(StatusCodes.InternalServerError, s"Error: $t")
+  }
+
+  val customHeaders = `Access-Control-Allow-Headers`("Content-Type, Authorization") ::
+    `Access-Control-Allow-Methods`(POST) ::
+    `Cache-Control`(public, `no-store`, `max-age`(0)) :: Nil
+
+  lazy val makeSocketHandler: Flow[Message, TextMessage.Strict, NotUsed] = {
+
+    // create a flow transforming a queue of string -> string
+    val (flowInput, flowOutput) = Source.queue[String](10, OverflowStrategy.dropTail).toMat(BroadcastHub.sink[String])(Keep.both).run()
+
+    // register an actor that feeds the queue when a payment is received
+    actorSystem.actorOf(Props(new Actor {
+      override def preStart: Unit = context.system.eventStream.subscribe(self, classOf[PaymentReceived])
+
+      def receive: Receive = {
+        case received: PaymentReceived => flowInput.offer(received.paymentHash.toString)
+      }
+    }))
+
+    Flow[Message]
+      .mapConcat(_ => Nil) // Ignore heartbeats and other data from the client
+      .merge(flowOutput) // Stream the data we want to the client
+      .map(TextMessage.apply)
+  }
+
+  val timeoutResponse: HttpRequest => HttpResponse = { r =>
+    HttpResponse(StatusCodes.RequestTimeout).withEntity(ContentTypes.`application/json`, "request timed out")
+  }
+
+  def userPassAuthenticator(credentials: Credentials): Future[Option[String]] = credentials match {
+    case p@Credentials.Provided(id) if p.verify(password) => Future.successful(Some(id))
+    case _ => akka.pattern.after(1 second, using = actorSystem.scheduler)(Future.successful(None)) // force a 1 sec pause to deter brute force
```

Comment:
This would allow removing a class-level field:
```suggestion
    case _ => akka.pattern.after(1 second, using = actorSystem.scheduler)(Future.successful(None))(actorSystem.dispatcher) // force a 1 sec pause to deter brute force
```

Full diff:

```diff
diff --git a/eclair-core/src/main/scala/fr/acinq/eclair/api/NewService.scala b/eclair-core/src/main/scala/fr/acinq/eclair/api/NewService.scala
index d07642e1..87d11b2d 100644
--- a/eclair-core/src/main/scala/fr/acinq/eclair/api/NewService.scala
+++ b/eclair-core/src/main/scala/fr/acinq/eclair/api/NewService.scala
@@ -33,7 +33,6 @@ trait NewService extends Directives with Logging with MetaService {
   def eclairApi: EclairApi

   implicit val actorSystem: ActorSystem
-  implicit lazy val ec = actorSystem.dispatcher
   implicit val mat: ActorMaterializer

   // a named and typed URL parameter used across several routes, 32-bytes hex-encoded
@@ -76,7 +75,7 @@ trait NewService extends Directives with Logging with MetaService {

   def userPassAuthenticator(credentials: Credentials): Future[Option[String]] = credentials match {
     case p@Credentials.Provided(id) if p.verify(password) => Future.successful(Some(id))
-    case _ => akka.pattern.after(1 second, using = actorSystem.scheduler)(Future.successful(None)) // force a 1 sec pause to deter brute force
+    case _ => akka.pattern.after(1 second, using = actorSystem.scheduler)(Future.successful(None))(actorSystem.dispatcher) // force a 1 sec pause to deter brute force
   }

   val route: Route = {
```

Reply:
Addressed in 7092989